src/middleware.ts

```diff
@@ -0,0 +1,41 @@
+import { runWithAmplifyServerContext } from '@/utils/amplifyServerUtils';
+
+import { fetchAuthSession } from 'aws-amplify/auth/server';
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function middleware(request: NextRequest) {
+  const response = NextResponse.next();
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call
```

Comment:
[This method's](https://github.com/aws-amplify/amplify-js/blob/04627618afe526dd776ee678b723ed6e697f440d/packages/adapter-nextjs/src/types/NextServer.ts#L86) type seems not to be open for external so I cannot help setting this ignore of eslint check for now.
[Example](https://docs.amplify.aws/nextjs/build-a-backend/server-side-rendering/nextjs-v5-to-v6-migration-guide/#step-2-server-side-configuration)